metadata-store-directory-common/src/main/java/org/apache/helix/msdcommon/datamodel/TrieRoutingData.java

```diff
@@ -167,8 +167,8 @@ private TrieNode getLongestPrefixNodeAlongPath(String path) {
   }
 
   /*
-   * Checks for the edge case when the only sharding key in provided routing data is the delimiter
-   * or an empty string. When this is the case, the trie is valid and contains only one node, which
+   * Checks for the edge case when the only sharding key in provided routing data is the delimiter.
+   * When this is the case, the trie is valid and contains only one node, which
```

Comment:
Question: we still want to throw an exception when the data is empty, right?

Reply:
The comment change here was regarding to the sharding key being either "/" or "", not routing data being empty. I believe your question is about when the entire routing data is empty: the answer is yes, we throw an exception. That behavior, as we agreed on, will not be changed.